### PR TITLE
Issue #4012: record offline-online smoke closure audit

### PR DESCRIPTION
**Pull request by ll7**
## Reviewer-critical summary

What changed: records the issue #4012 closure audit against merged PRs #4170 and #4472, reruns the CPU-only paired smoke, and promotes a compact evidence summary for the completed offline-to-online and from-scratch Soft Actor-Critic arms.

Why now: the latest issue state kept #4012 open after #4472 because the code path existed but the paired smoke output had not been captured as durable review evidence. This PR is the smallest remaining integration slice, not another guardrail packet.

Claim boundary: diagnostic-smoke workflow evidence only. The smoke proves the offline-to-online lane and matched scratch arm can complete at tiny CPU scale; it is not benchmark evidence and does not support any robustness, dissertation, or paper-facing performance claim.

Required validation:

- `scripts/dev/run_worktree_shared_venv.sh -- uv run pytest tests/training/test_offline_online_rl.py tests/training/test_hybrid_replay_buffer.py tests/training/test_train_sac_sb3_offline_online.py tests/training/test_run_offline_online_rl.py -q` -> 24 passed.
- `scripts/dev/run_worktree_shared_venv.sh -- uv run --extra training python scripts/training/run_offline_online_rl.py --config configs/training/offline_online_rl/issue_4012_sac_smoke.yaml` -> completed both `offline_online` and `scratch` arms.
- `python3 -m json.tool docs/context/evidence/issue_4012_offline_online_smoke_2026-07-05/summary.json` -> passed.
- `git diff --check` -> passed.
- `PR_READY_MODE=final BASE_REF=origin/main PR_READY_PR_BODY_FILE=/home/luttkule/git/robot_sf_ll7/.git/codex-agent-runs/active/issue-4012/PR_BODY.md scripts/dev/pr_ready_check.sh` -> passed; 1895 tests passed.

Remaining blocker: none for the issue #4012 diagnostic-smoke acceptance criteria.

State propagation: no separate issue comment is posted because this task explicitly has `comment_issue_or_pr: false`. This PR body and `docs/context/issue_4012_offline_online_rl.md` are the durable state surfaces for this slice.

## Contract delta

- New schema fields: none.
- New fail-closed blockers: none; this PR documents existing #4170/#4472 fail-closed behavior and the completed smoke evidence.
- Compatibility impact: docs/evidence only. No runtime behavior, configs, training defaults, or benchmark semantics change.
- Evidence artifact: `docs/context/evidence/issue_4012_offline_online_smoke_2026-07-05/summary.json` records compact checksums and paired-arm status; large `output/` checkpoints remain ignored local diagnostic artifacts.

## Issue

Issue link: #4012. This PR provides the criterion-to-evidence closure audit and the missing diagnostic smoke evidence; the reviewer can close the issue after accepting the claim boundary.

## Scope summary

- Updates `docs/context/issue_4012_offline_online_rl.md` with a criterion-to-evidence closure table.
- Adds compact evidence for the 2026-07-05 CPU paired smoke run.

## Explicitly out of scope

- No full benchmark campaign run.
- No Slurm/GPU submission.
- No paper/dissertation claim edits.
- No issue comments, merges, releases, deletions, or private queue-routing state encoded in tracked files.

## Acceptance evidence

<details>
<summary>Acceptance evidence</summary>

| Criterion | Evidence |
| --- | --- |
| Offline-pretrained policy initializes online fine-tuning end-to-end on smoke scenario. | PR #4170 added the lane; the 2026-07-05 smoke completed `offline_online` with `offline_online_enabled=true` and 256 online timesteps. |
| Comparison vs from-scratch online training. | PR #4170 added the scratch arm; PR #4472 added paired reporting; the 2026-07-05 smoke completed both arms under seed 4012. |
| Claim boundary documented. | The context note and evidence summary set diagnostic-smoke-only, `eligible_for_claim=false`, and no paper-facing claim. |
| Detailed implementation-plan criteria. | Mapped in `docs/context/issue_4012_offline_online_rl.md`; focused tests passed for transition loading, preflight, hybrid replay, config behavior, fail-closed reporting, and orchestrator behavior. |

</details>

## Summary by CodeRabbit

* **Documentation**
  * Updated the offline-to-online reinforcement learning guide with clearer CPU-only diagnostic smoke-test steps, required inputs (including dataset/environment preflight), and output-location guidance.
  * Added a closure audit section mapping acceptance criteria to the specific evidence and smoke-run results, including diagnostic-only framing and artifact disposition rules.
  * Updated catalog coverage to include the new evidence summary for the issue 4012 offline/online scratch smoke lane.
* **Bug Fixes**
  * Clarified diagnostic smoke-run boundaries and evidence handling to reduce confusion between temporary local outputs and durable promoted results.

### CODE REVIEW

**Comment by ll7** [comment]
## Gate review (Opus 4.8 review gate) — head `f5ba19034` → fix pushed `52a6b5ee5`

Docs/evidence-only closure-audit slice for #4012 (+142/-15, 2 files: `summary.json` + `issue_4012_offline_online_rl.md`). Claim boundary is diagnostic-smoke-only, correctly stated. Reviewed the full diff.

### Open review-bot threads (exhaustive)
1. **[CodeRabbit — Major, Data Integrity]** "Register this evidence file in `docs/context/catalog.yaml`." → **FIXED (phase 2).** `summary.json` was unregistered, which is exactly what failed the `docs-evidence-integrity` CI check (`evidence file is not registered in docs/context/catalog.yaml`). Added the entry (`status: evidence`, `freshness: evidence`, `area: workflow_evidence`); local `scripts/dev/check_docs_evidence_integrity.py` now reports "2 changed file(s) passed."
2. **[CodeRabbit — Minor, Maintainability]** "Define `RLTrajectoryDataset.v1` on first use." → **FIXED (phase 2).** Added a concise inline definition at first use (versioned JSONL schema for recorded RL trajectory rows — obs/action/reward/next-obs transitions — consumed as offline training data).

### Findings from the diff
1. **[VERIFIED — root-cause of UNSTABLE]** The only failing check was `docs-evidence-integrity`, caused solely by the missing catalog registration (thread #1). Fixed; no other integrity gap.
2. **[VERIFIED — claim discipline sound]** `summary.json` records `evidence_tier: diagnostic-smoke-only`, `claim_boundary: "Pipeline execution only; not benchmark evidence"`, `forbidden_actions_confirmed` all false (no compute/merge/release/delete), and a `tracked_policy` that keeps only the compact summary while large checkpoints stay in ignored `output/`. Per-artifact SHA-256 recorded. Both arms (`offline_online`, `scratch`) report `status: completed`, `blocker: null`.
3. **[VERIFIED — closure scope correct]** PR body uses "Refs"-style framing; the doc's acceptance table maps each #4012 criterion to merged PRs #4170/#4472 with dated evidence. "Remaining blocker: none for the diagnostic-smoke acceptance criteria." No deferred item lacks a tracking anchor.

### Disposition
Both review threads fixed in-scope; no HIGH/MEDIUM correctness findings. Merging once the re-run CI (docs-evidence-integrity + pr-body-contracts) is green on `52a6b5ee5`.
